src/pipeline.rs

```diff
@@ -572,45 +577,85 @@ pub(crate) async fn pipeline_task(
                 is_transactional,
             }) => {
-                info!("{ctx} Connecting {endpoint} to port 'in'");
+                info!("{ctx} Connecting '{endpoint}' to port '{port}'");
+                if !input_does_exist(&port, &pipeline) {
+                    error!("{ctx} Error connecting input pipeline '{port}' as it does not exist",);
+                    if tx
+                        .send(Err("input port doesn't exist".into()))
+                        .await
+                        .is_err()
+                    {
+                        error!("{ctx} Error sending status report.");
+                    }
+                    continue;
+                }
                 inputs.insert(endpoint, (is_transactional, target));
+                if tx.send(Ok(())).await.is_err() {
+                    error!("{ctx} Status report sent.");
+                }
             }
+
             AnyMsg::Mgmt(MgmtMsg::ConnectOutput {
                 port,
                 endpoint,
                 target,
+                tx,
             }) => {
                 info!("{ctx} Connecting port '{port}' to {endpoint}",);
                 // add error statement for port out in pipeline, currently no error messages
                 if output_doesnt_exist(&port, &pipeline) {
                     error!("{ctx} Error connecting output pipeline {port}");
+                    if tx
+                        .send(Err("output port doesn't exist".into()))
+                        .await
+                        .is_err()
+                    {
+                        error!("{ctx} Error sending status report.");
+                    }
                     continue;
                 }
                 // notify other pipeline about a new input
-                if let OutputTarget::Pipeline(pipe) = &target {
-                    // avoid linking the same pipeline as input to itself
-                    // as this will create a nasty circle filling up queues.
-                    // In general this does not avoid cycles via more complex constructs.
-                    //
-                    if id.pipeline_alias() == endpoint.alias() {
-                        if let Err(e) = pipe
-                            .send_mgmt(MgmtMsg::ConnectInput {
-                                endpoint: DeployEndpoint::new(
-                                    id.pipeline_alias(),
-                                    &port,
-                                    endpoint.meta(),
-                                ),
-                                target: InputTarget::Pipeline(Box::new(addr.clone())),
-                                is_transactional: true,
-                            })
-                            .await
-                        {
-                            error!("{ctx} Error connecting input pipeline {endpoint}: {e}",);
-                        }
-                    }
+                // if let OutputTarget::Pipeline(pipe) = &target {
```

Comment:
why is this commented out?

Reply:
Then we can scratch this variant, it seems it doesnt do anything, does it?

Reply:
Ja the entire if block that's commented out can go 👍 we'll clean it up